seed/seedwriter/writer.go

```diff
@@ -466,9 +466,26 @@ func IsSytemDirectoryExistsError(err error) bool {
 	return ok
 }
 
+// correctedValidationSetAtSequence returns the corrected AtSequence for an
+// validation set. If any restrictions have been set in
+// in the manifest, then we must use the sequence and pinning status from
+// that instead of whats set in the model.
+func (w *Writer) correctedValidationSetAtSequence(vsm *asserts.ModelValidationSet) *asserts.AtSequence {
+	atSeq := vsm.AtSequence()
+	for _, vs := range w.manifest.AllowedValidationSets() {
+		if vs.AccountID == vsm.AccountID && vs.Name == vsm.Name {
+			atSeq.Sequence = vs.Sequence
```

Comment:
the manifest can change sequence only for things not pinned in the model, if the validation sequence is pinned in the model the manifest can only have the same info otherwise it's an error?

Reply:
That actually makes a lot of sense, it's definitely a bit to flexible. I added a check for this and also a separate unit test for this case.